libcudacxx/include/cuda/std/__exception/cuda_error.h

```diff
@@ -27,41 +27,54 @@
 #endif // !_CCCL_CUDA_COMPILER_NVCC && !_CCCL_CUDA_COMPILER_NVHPC
 
 #include <cuda/std/__exception/terminate.h>
-#include <cuda/std/detail/libcxx/include/stdexcept>
 
-#include <nv/target>
 #if !defined(_CCCL_COMPILER_NVRTC)
 #  include <cstdio>
+#  include <stdexcept>
 #endif // !_CCCL_COMPILER_NVRTC
 
+#include <nv/target>
+
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
 
 /**
  * @brief Exception thrown when a CUDA error is encountered.
  */
-class cuda_error : public _CUDA_VSTD_NOVERSION::runtime_error
+
+#ifndef _LIBCUDACXX_NO_EXCEPTIONS
+class cuda_error : public ::std::runtime_error
 {
-public:
-  cuda_error(::cudaError_t __status, const char* __msg) noexcept
-      : _CUDA_VSTD_NOVERSION::runtime_error("")
+private:
+  static char* __format_string(::cudaError_t __status, const char* __msg, char __formatted_msg[256] = {}) noexcept
   {
-    ::snprintf(
-      const_cast<char*>(this->what()), _CUDA_VSTD::__libcpp_refstring::__length, "cudaError %d: %s", __status, __msg);
+    ::snprintf(__formatted_msg, 256, "cudaError %d: %s", __status, __msg);
+    return __formatted_msg;
   }
```

Comment:
Question: So this does not return a dangling pointer, because the `char __formatted_msg[256] = {}` creates the char array in the caller's scope, where it lives until the end of the full expression, at which point the returned char pointer is already consumed by the constructor of `std::runtime_error`?

Reply:
That was my intent, now that you say it I am not sure, whether this is actually correct.

movign to a function local static

Reply:
Thinking a second time about it, it probably isn't correct. The `char __formatted_msg[256] = {}` is actually `char* __formatted_msg = {}`, so a null pointer by default :D I think it would work if you changed it to `::cuda::std::array<char, 256> = {}`.